### ISSUE TITLE
pgbackrest is not working on standby

### ISSUE THREAD

**Issue by moindba** [pgbackrest/pgbackrest#1459]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version:
2.33
2. PostgreSQL version:
11.11
3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each:
CentOS 7.9
4. Did you install pgBackRest from source or from a package?
package

5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
    - errors in the postgresql log file before or during the time you experienced the issue
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

7. Describe the issue:
I can take backups from Master DB, but if i want to take backups from standby DB i couldn't able to do so.

Kindly help me how i can successfully take backups from standby DB.

[pgbackrest_standby_issue.txt](https://github.com/pgbackrest/pgbackrest/files/6803037/pgbackrest_standby_issue.txt)

**Comment by pgstef** [comment]
Hi,

Your 2 servers should be part of the same `stanza` and have access to the same repository.
That's where your issue comes from. You use `pg1` and `pg2` as different stanzas, where you should use the same. And make sure `/S01/pgsql/pgbackrest` is a shared location between both servers...

The goal is that the primary sends WAL archives to the repo, and the standby sees it coming. Until your `check` command is working, you probably won't be able to take backup-standby.

Examples:
- https://pgbackrest.org/user-guide.html#standby-backup
- https://pgstef.github.io/2020/09/04/combining_pgbackrest_and_streaming_replication.html

Kind regards

**Comment by moindba** [comment]
Thanks for your reply !

I case if i dont have shared backup location between primary and standby then what will be the work around ?

**Comment by pgstef** [comment]
> I case if i dont have shared backup location between primary and standby then what will be the work around ?

That's the all point, the PG servers (within the same replication cluster) should share a repo. Also, a very important point is that you should **never** keep your backups on the same server as your PG data.

In case of replication, I really like using a [dedicated backup server](https://pgstef.github.io/2020/11/21/combining_pgbackrest_dedicated_repository_host_and_streaming_replication.html). 

What happens when you promote your standby as new primary ? You can't take "backup-standby" from the same host anymore and you have to manually reconstruct your old primary... Having a dedicated backup server would detect automatically which host is primary/standby and pull the backup from the accurate server. And furthermore, since both PG hosts will share the same repo, you could use the backups to refresh your old primary and make it standby...

Anyway, long story short, you should share the repo, there are so many ways to achieve this (cloud storage, internal MinIO, nfs, cifs, dedicated backup server,...).

To be honest, a corner case that could work in theory is to enable `archive_mode` to `always`, asking the standby server to push archives and disabling pgbackrest [archive-mode-check](https://pgbackrest.org/configuration.html#section-backup/option-archive-mode-check). But then you'd have to disable it on your primary or the archives generated/pushed to the primary repo will never be expired (given wal archives are expired based on backups present in the same repo). So, unless you really know what you're doing, you shouldn't do this, and keep it simple (using a shared repo).
